compiler/rustc_errors/src/styled_buffer.rs

```diff
@@ -49,29 +62,25 @@ impl StyledBuffer {
     }
 
     fn ensure_lines(&mut self, line: usize) {
-        while line >= self.text.len() {
-            self.text.push(vec![]);
-            self.styles.push(vec![]);
+        if line >= self.lines.len() {
+            self.lines.resize_with(line + 1, || vec![]);
```

Comment:
```suggestion
            self.lines.resize(line + 1, Vec::new());
```
this is a really small thing, but i quite strongly dislike `resize_with(line + 1, || vec![])`.

`Vec::new()` doesn't allocate and should be fully evaluated at compile time. So there is no perf benefit to using `resize_with` and `resize` is easier to read